zbus_macros/src/iface.rs

```diff
@@ -1051,37 +1053,38 @@ fn get_args_from_inputs(
 
                 let signal_context_arg = &input.pat;
 
-                signal_emitter_arg_decl = Some(quote! {
-                    let #signal_context_arg = match hdr.path() {
-                        ::std::option::Option::Some(p) => {
-                            #zbus::object_server::SignalEmitter::new(connection, p).expect("Infallible conversion failed")
-                        }
-                        ::std::option::Option::None => {
-                            let err = #zbus::fdo::Error::UnknownObject("Path Required".into());
-                            return connection.reply_dbus_error(&hdr, err).await;
-                        }
-                    };
-                });
+                signal_emitter_arg_decl = match method_type {
+                    MethodType::Property(PropertyType::Getter) => {
+                        Some(quote! { let #signal_context_arg = signal_emitter.clone(); })
+                    }
+                    _ => Some(quote! {
+                        let #signal_context_arg = match hdr.path() {
+                            ::std::option::Option::Some(p) => {
+                                #zbus::object_server::SignalEmitter::new(connection, p).expect("Infallible conversion failed")
+                            }
+                            ::std::option::Option::None => {
```

Comment:
Hmm.. isn't this path going to be taken for the property setter too? If so, we don't use the `signal_emitter` arg to `Interface::set*`. I think it'd be best to do the same for getter and setter here.

Reply:
Crap, you're right, sorry and good catch! I'll have a think about how that came about, then I'll update the PR.

Reply:
Not sure how I ended up doing that, it's definitely not intended. I guess that was just part of a larger refactoring effort that just ended up in a typo there. Should be OK now.